src/software/abinit.jl

```diff
@@ -962,3 +962,29 @@ function write_abinit_modes(modes::Array{SVector{6,Float64}}, energies::Vector{F
         end
     end
 end
+
+"""
+    read_abinit_anaddb_PHDOS(filename::AbstractString) -> Tuple{DensityOfStates, Vector{ProjectedDensityOfStates}}
+Reads the PHDOS file from ABINIT's anaddb script.
+"""
+function read_abinit_anaddb_PHDOS(filename::AbstractString)
+    data = readlines(filename)
+    # Skip header (lines 1-8)
+    data = data[9:length(data)]
+    # Initialize matrix to store data
+    # Energy | total PHDOS | int PHDOS | AtomType 1 PHDOS | AtomType 1 intPHDOS | ...
+    num_col = length(split(data[1]))
+    data_new = Matrix{Float64}(undef,length(data),num_col)
+    # Read in data
+    for i in 1:length(data)
+        data_new[i,:] = parse.(Float64, split(data[i]))
+    end
```

Comment:
You can convert this into a list comprehension, so you don't have to start with initializing a matrix. It cuts down on the number of lines, and is generally the fastest way of building a matrix.
```julia
data_new = [parse(Float64, split(data[i])) for i in eachindex(data), j in 1:num_col]
```